### PR TITLE
Assorted minor fixes

### PR DESCRIPTION
**Pull request by kgaillot**
This is a variety of largely insignificant things I ran across when doing other projects. Some log message improvements are the closest to being useful, along with a Y2038 fix.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , another low priority one, but can you review when you get a chance?

File: tools/crm_resource.c [hunk 1]
```diff
@@ -39,28 +39,34 @@ static pe_working_set_t *data_set = NULL;
...
+        pcmk_controld_api_t *save_controld_api = controld_api;
+
+        controld_api = NULL;
+        pcmk_free_controld_api(save_controld_api);
     }
     pe_free_working_set(data_set);
     data_set = NULL;
     crm_exit(exit_code);
-    return;
+    return exit_code;
```
Comment (clumens) [inline tools/crm_resource.c]: An alternate might be to mark this function as _Noreturn.  We do that in a couple places already.  It doesn't really matter to me one way or the other.

File: tools/crm_mon.c [hunk 2]
```diff
@@ -775,14 +775,25 @@ cib_connect(gboolean full)
 
     if (cib->state != cib_connected_query && cib->state != cib_connected_command) {
         crm_trace("Connecting to the CIB");
+
+        /* Hack: the CIB signon will print the prompt for a password if needed,
+         * but to stderr. If we're in curses, show it on the screen instead.
+         *
+         * @TODO Add a password prompt (maybe including input) function to
+         *       pcmk__output_t and use it in libcib.
+         */
```
Comment (clumens) [inline tools/crm_mon.c]: That would be nice.  While trying to get rid of output_format in crm_mon as much as possible, this one always caused me problems.

File: tools/crm_resource.c (hunk 1 above)
Comment (kgaillot) [reply tools/crm_resource.c]: Except it's not now :) which is the point (it can return at the beginning)

File: tools/crm_mon.c (hunk 2 above)
Comment (kgaillot) [reply tools/crm_mon.c]: Yep, it would help abstract away the terminal garbage for plain text too

Comment (kgaillot) [reply tools/crm_mon.c]: (in cib_remote_signon(), but obviously only once everything's on the new output model)

**Pull request merged**